### PR TITLE
[Merged by Bors] - feat(Algebra/Order/BigOperators): prove lemmas for `List.prod` and `Multiset.prod` on `CommMonoidWithZero`

### PR DESCRIPTION
**Pull request by Command-Master**
Also relax typeclass assumptions on `prod_nonneg`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by YaelDillies — COMMENTED** [review]
Those lemmas now look misplaced. Can you move them to a new file `Algebra.Order.BigOperators.GroupWithZero`?

**Comment by Command-Master** [comment]
> Those lemmas now look misplaced. Can you move them to a new file `Algebra.Order.BigOperators.GroupWithZero`?

The corresponding lemmas for Finset are also in `Ring`, for instance [Finset.prod_nonneg](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Order/BigOperators/Ring/Finset.html#Finset.prod_nonneg)

**Comment by YaelDillies** [comment]
That is an oversight. They should be moved too (although no need for you to do it in the current PR)

File: Mathlib/Algebra/Order/BigOperators/GroupWithZero/Multiset.lean [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/-
+Copyright (c) 2024 Daniel Weber. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniel Weber
```
Comment (YaelDillies) [inline Mathlib/Algebra/Order/BigOperators/GroupWithZero/Multiset.lean:4]: `Multiset.prod_nonneg` dates from https://github.com/leanprover-community/mathlib3/pull/8787

File: Mathlib/Algebra/Order/BigOperators/GroupWithZero/List.lean [hunk 2]
```diff
@@ -0,0 +1,99 @@
+/-
+Copyright (c) 2024 Daniel Weber. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniel Weber
```
Comment (YaelDillies) [inline Mathlib/Algebra/Order/BigOperators/GroupWithZero/List.lean:4]: `List.prod_pos` dates from https://github.com/leanprover-community/mathlib3/pull/10184

File: Mathlib/Algebra/Order/BigOperators/GroupWithZero/Multiset.lean (hunk 1 above)
Comment (Command-Master) [reply Mathlib/Algebra/Order/BigOperators/GroupWithZero/Multiset.lean]: Sorry about that, is this better?

Comment (YaelDillies) [reply Mathlib/Algebra/Order/BigOperators/GroupWithZero/Multiset.lean]: Looks better, but you credited the wrong person for this file! The other file is good

Comment (Command-Master) [reply Mathlib/Algebra/Order/BigOperators/GroupWithZero/Multiset.lean]: Oops

**Comment by YaelDillies** [comment]
Thanks :)

maintainer merge

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**